client/app/components/onboardingGuide/OnboardingGuide.jsx

```diff
@@ -0,0 +1,183 @@
+      // null state means that page component's root path is reached and
+      // previous page is actually on Rails side - i.e. one step further
+      window.history.back();
+    }
+  }
+
+  handlePageChange(path) {
+    this.props.actions.updateGuidePage(path, true);
+  }
+
+  render() {
+    const { Page, translations, ...opts } = selectChild(this.props.data, this.nextStep);
+    return (
+      <Page
+        changePage={ this.handlePageChange }
+        initialPath={ this.initialPath }
+        name={ this.props.data.name }
+        t={ translate(translations) }
+        { ...opts } />
+    );
+  }
+};
+
+// Select child component (page/view) to be rendered
+// Returns object (including child component) based on props.data & nextStep
+const selectChild = function(data, nextStep) {
+  const { path, onboarding_data, translations } = data;
+  const pageData = (path.length > 0) ? onboarding_data[path.substring(1)] : {};
+  const commonTranslations = { back_to_todo: translations.back_to_todo };
+
+  switch(path) {
+    case "/slogan_and_description":
+      return  { Page: GuideSloganAndDescriptionPage,
+                pageData,
+                translations: Object.assign({}, commonTranslations, translations.slogan_and_description)
+              };
+    case "/cover_photo":
+      return  { Page: GuideCoverPhotoPage,
+                pageData,
+                translations: Object.assign({}, commonTranslations, translations.cover_photo)
+              };
+    case "/filter":
+      return  { Page: GuideFilterPage,
+                pageData,
+                translations: Object.assign({}, commonTranslations, translations.filter)
+              };
+    case "/paypal":
+      return  { Page: GuidePaypalPage,
+                pageData,
+                translations: Object.assign({}, commonTranslations, translations.paypal)
+              };
+    case "/listing":
+      return  { Page: GuideListingPage,
+                pageData,
+                translations: Object.assign({}, commonTranslations, translations.listing)
+              };
+    case "/invitation":
+      return  { Page: GuideInvitationPage,
+                pageData,
+                translations: Object.assign({}, commonTranslations, translations.invitation)
+              };
+    default:
+      return  { Page: GuideStatusPage,
+                onboarding_data,
+                translations: Object.assign({}, commonTranslations, translations.status_page),
+                nextStep
+              };
+  }
+}
+
+// Get curried function translate page related translations
+const translate = function(translations) {
+  return function(translationKey) {
+    return translations[translationKey];
+  }
+}
+
+// Get link and title of next recommended onboarding step
+const nextStep = function(data, translate, initialPath) {
+
+  const stepOrder = [
+    'slogan_and_description',
+    'cover_photo',
+    'filter',
+    'paypal',
+    'listing',
+    'invitation'
```

Comment:
There was long discussions with Olli that who should own the state of these steps. I was originally leaning towards Rails having something to say about next step (since we have separate frontend components instead of SPA), but I decided to go with this direction in the end. So, it's (web) applications job to decide the order, which means hard coded values need to exist somewhere.